integrations/GATEWAYS_FILE.md

```diff
@@ -0,0 +1,91 @@
+# IPIP 0280: Gateways file
+- Start Date: 2022-12-08
+- Related Issues:
+  - https://github.com/ipfs/kubo/issues/8847
+
+## Summary
+
+Defines how IPFS implementations must expose their gateway. This only affects the file-based method of exposiing a gateway.
+
+## Motivation
+
+Applications wanting to use IPFS resources are, without this spec, left to invent their own ways of finding a gateway. This spec defines how an application wanting to implement IPFS support can find a list of gateways.
+
+Users should be aware that this is just a list of gateways. It could contain a local gateway but doesn't have to. It's up to the user to do filtering on that list to find an actual local gateway.
+
+## Detailed design
+
+This integration spec defines the recommended way for IPFS implementations to expose the gateway they host for IPFS integrations (think applications wanting to support IPFS) to find and use said gateways.
+
+An IPFS implementation must expose the gateway it serves in a file called `gateways` (more on where this file is located in the section below). The gateway must be exposed as a single line in the following format:
+
+`http://<ip>:<port>`
+
+In case the `gateways` file already exists, this line must be **prepended** with the rest of the data left as-is. This is a soft guarantee that any locally running gateways are at the top of the list. In any other case the `gateways` file must be created and the `gateway` line inserted to it as one line terminated with `\r\n`.
+
+Upon shutdown of the IPFS implementation it is to remove it's specific gateway line from the `gateways` file and leave no gaps. It must take care potential empty lines and leave a file behind with the remaning gateways each on a single line and terminated - per line - with `\r\n`.
+
+### Gateways file placement
+
+As a reference, [this](https://github.com/cjbassi/platform-dirs-rs#path-list) list of configuration folders is used.
+
+| | with variables | full paths |
+| -------- | -------- | -------- |
+| (user) Windows     | %APPDATA%/ipfs     | C:/Users/%USERNAME%/AppData/Roaming/ipfs     |
+| (global) Windows     | %PROGRAMDATA%/ipfs     | C:/ProgramData/ipfs     |
+| (user) macOS     | $XDG_CONFIG_HOME/ipfs     | ~/Library/Application Support/ipfs     |
+| (global) macOS     | N/A     | /Library/Application Support/ipfs     |
+| (user) Linux     | $XDG_CONFIG_HOME/ipfs     | ~/.config/ipfs     |
+| (global) Linux     | N/A     | /etc/ipfs     |
```

Comment:
Nit: missing final ` | ` on last line of table. May cause rendering issues in future